support/lints/error_suffix/src/lib.rs

```diff
@@ -16,10 +16,15 @@ dylint_linting::declare_late_lint! {
 
 impl<'tcx> LateLintPass<'tcx> for ErrorSuffix {
     fn check_item(&mut self, cx: &rustc_lint::LateContext<'tcx>, item: &'tcx Item<'tcx>) {
-        let item_name = item.ident.name.as_str();
+        let ident = match item.kind {
+            ItemKind::Enum(ident, ..) | ItemKind::Struct(ident, ..) => ident,
+            _ => return,
+        };
+
+        let item_name = ident.name.as_str();
 
         match &item.kind {
-            ItemKind::Enum(_, _) | ItemKind::Struct(_, _) => {
+            ItemKind::Enum(_, _, _) | ItemKind::Struct(_, _, _) => {
```

Comment:
In these cases we could probably match against `ItemKind::Enum(..)` which would work even if the field count was updated again, but not a big deal either way